### DISCUSSION TITLE
Using CuTe for linalg

### DISCUSSION THREAD

**Discussion by ASKabalan** [NVIDIA/cutlass#1456] [Q&A] [answered]
Hello cutlass and cute team.
Does it make sense to use cute without cutlass.
I just want to do some linear algebra and some algorithms on structured data  (maybe some gemms). Data as in matrices tensors and spherical coordinates (no deep learning)

For example in [tensor_algorithms](https://github.com/NVIDIA/cutlass/blob/main/include%2Fcute%2Falgorithm%2Ftensor_algorithms.hpp)

I imagine that these algorithms are implicit cuda kernels.
Do they automatically target hopper if I build with the proper cuda architecture?

Thank you very much.

**Comment by hwu36** [comment, marked as answer]
> Does it make sense to use cute without cutlass.

yes

> Do they automatically target hopper if I build with the proper cuda architecture?

yes
